pkg/deploy/lattice/access_log_subscription_manager.go

```diff
@@ -0,0 +1,106 @@
+package lattice
+
+import (
+	"context"
+
+	"github.com/aws/aws-sdk-go/service/vpclattice"
+
+	"github.com/aws/aws-application-networking-k8s/pkg/aws"
+	"github.com/aws/aws-application-networking-k8s/pkg/aws/services"
+	"github.com/aws/aws-application-networking-k8s/pkg/model/lattice"
+	"github.com/aws/aws-application-networking-k8s/pkg/utils/gwlog"
+)
+
+//go:generate mockgen -destination access_log_subscription_manager_mock.go -package lattice github.com/aws/aws-application-networking-k8s/pkg/deploy/lattice AccessLogSubscriptionManager
+
+type AccessLogSubscriptionManager interface {
+	Create(ctx context.Context, accessLogSubscription *lattice.AccessLogSubscription) (*lattice.AccessLogSubscriptionStatus, error)
+}
+
+type defaultAccessLogSubscriptionManager struct {
+	log   gwlog.Logger
+	cloud aws.Cloud
+}
+
+func NewAccessLogSubscriptionManager(
+	log gwlog.Logger,
+	cloud aws.Cloud,
+) *defaultAccessLogSubscriptionManager {
+	return &defaultAccessLogSubscriptionManager{
+		log:   log,
+		cloud: cloud,
+	}
+}
+
+func (m *defaultAccessLogSubscriptionManager) Create(
+	ctx context.Context,
+	accessLogSubscription *lattice.AccessLogSubscription,
+) (*lattice.AccessLogSubscriptionStatus, error) {
+	vpcLatticeSess := m.cloud.Lattice()
+
+	var resourceIdentifier string
+	if accessLogSubscription.Spec.SourceType == lattice.ServiceNetworkSourceType {
```

Comment:
Sure, done next commit!
Note that I'll be using fmt.Errorf rather than NewInvalidError. This would be a case where the controller is improperly handling something, rather than the user input being invalid.